### DISCUSSION TITLE
Yoda integration with Research Drive

### DISCUSSION THREAD

**Discussion by MVersteijlen** [UtrechtUniversity/yoda#208] [Ideas]
Scenario:

Researcher collaborates on data with peers both inside and outside the TU/e. For this purpose they use Research Drive as their preferred storage during the active research phase, as it provides safe (as in backed up regularly) storage with the added benefit of the ease to share the folder(s) containing the dataset(s) with their peers. Once the actual research is done, they want to have the corresponding dataset(s) archived, in order to fulfill the obligations set by a.o. the grant provider, but also to make it easier to reuse their data in further research. Ideally the action of moving the data from Research Drive to the data archive in use by the TU/e should be trivial. A possibility to move a dataset (folder?) from Research Drive to Yoda with a simple mouse click would greatly lighten the burden of the archival process for the researcher.

User stories:

As a researcher I want to be able to archive my datasets so that I fulfill the obligations set by a.o. the government and grant providers, and to provide for easier reuse of said datasets. Ideally the process of moving the data from the working/collaborating environment to this data archive should be trivial.

Possible solution:

An extra option in the drop down menu next to a folder (...) could be created to copy/move the data to Yoda. Since both Research Drive and Yoda at the TU/e use SurfConext for authentication, they share a common user identifier. It should therefore be possible for Research Drive to check which groups in Yoda the user has write access to. If the logged in user has access to more than one group in Yoda, a pop up could ask which of these groups the user wants to archive the folder to. No extra functionality would be necessary in Research Drive for the addition of metadata to the dataset, since according to Yoda's business rules it's perfectly fine to have a non-metadated dataset in the research area, as long as the required metadata is added before submitting the dataset to the vault.

**Comment by lwesterhof** [comment]
Is this an idea for Yoda or for Research Drive? The possible solution consist of changes in Research Drive?

Follow up question, what is preventing researchers  to use Yoda as storage during the active research phase?

**Reply by MVersteijlen** [reply]
You're right, this may be more appropriate for the backlog of Research Drive. Do they have a similar page that we can post ideas?

The research storage on Yoda is less user friendly than for Research Drive. RD offers collaborative features with externals, and continuous WebDav functionality, so that researchers may use owncloud or similar to sync folders

**Reply by lwesterhof** [reply]
@aal-ssara can probably point us in the right direction for Research Drive ideas?

Yoda also offers collaborative features with externals and continuous WebDav functionality. Maybe some problems can be solved with configuration of you Yoda instance?

**Reply by a-surfer** [reply]
Hi @MVersteijlen ,

Since you mention obligations, I would like to start by stating that next to a technical solution as you request, you may need to adhere to your faculty/university/discipline/group's policy and best practices. These policy and best practices may state that you want to have data in your institute's Yoda instance and/or external repositories, depending on the characteristics of the data. So I see value in looking at both archival and publishing paths on both Yoda and Research Drive.

At SURF, together with our partners, we are working on integrating both archiving and publishing facilities in Yoda and in Research Drive. And we are looking as well into integrating Research Drive and Yoda to be able to have data flows covering both services. 

For everything related to Yoda, we are channeling requests through the forum that we call RDM TEC. The TU/e has representatives on this forum, so they are already in a good position to talk to you and support this case. I will pass your name on to them.

For Research Drive, you can get in contact with the team through your usual support channels, and through helpdesk@surfsara.nl .

**Comment by EmiliaJarochowska** [comment]
Hi,

I would like to follow up on this question. SURF Research Drive allows connecting external storage so I connected it via WebDAV. Initially the mounting of a Yoda volume on the Research Drive works, but after a few minutes the connection is lost. It appears mounted, but cannot be accessed. This can be solved by going on the Research Drive web interface and remounting the Yoda volume, but the connection is lost again. 

I won't go into the discussion here why to use both. The important point is that **there exists a simple technical possibility to copy directly between them**, but possibly with some quirk of configuration or a bug that makes them disconnect. Hopefully, once resolved, this could be a solution to @MVersteijlen' question?

**Comment by Danny-dK** [comment]
@MVersteijlen  Is this one still open or has contact been initiated with SURF? (otherwise an answer can be chosen and discussion moved to the closed section).

Note that regarding point of @EmiliaJarochowska, you can also use cyberduck to connect to both reserach drive and Yoda. When you have two cyberduck windows open, you can simply copy-paste (or drag and drop) between the different instances.

**Comment by Danny-dK** [comment]
No response. Discussion mark as answered and closed? @lwesterhof
